langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java

```diff
@@ -42,16 +44,25 @@ public static BTypedesc getElementType(Object td) {
     }
 
     private static BTypedesc getTypeDescValue(Type type) {
-        if (type.getTag() == TypeTags.ARRAY_TAG) {
-            return ValueCreator.createTypedescValue(((ArrayType) type).getElementType());
-        } else if (type.getTag() == TypeTags.STREAM_TAG) {
-            return ValueCreator.createTypedescValue(((StreamType) type).getConstrainedType());
-        } else if (type.getTag() == TypeTags.TABLE_TAG) {
-            return ValueCreator.createTypedescValue(((TableType) type).getConstrainedType());
-        } else if (type.getTag() == TypeTags.FINITE_TYPE_TAG && ((FiniteType) type).getValueSpace().size() == 1) {
-            return getTypeDescValue(((BValue) (((FiniteType) type).getValueSpace().iterator().next())).getType());
+        switch (type.getTag()) {
+            case TypeTags.ARRAY_TAG:
+                return ValueCreator.createTypedescValue(((ArrayType) type).getElementType());
+            case TypeTags.TUPLE_TAG:
+                return ValueCreator.createTypedescValue(
+                        TypeCreator.createUnionType(((TupleType) type).getTupleTypes()));
+            case TypeTags.STREAM_TAG:
+                return ValueCreator.createTypedescValue(((StreamType) type).getConstrainedType());
+            case TypeTags.TABLE_TAG:
+                return ValueCreator.createTypedescValue(((TableType) type).getConstrainedType());
```

Comment:
Lets write test for tables as well.

Reply:
I will look into this.

Reply:
It seems a ballerina code which covers the table case cannot be written. Even for the below code, this did not get covered. @pcnfernando 
```bal
table<map<string>> & readonly tbl = table [
         {a: "x", b: "3"},
         {a: "z", b: "5"}
];

function testToStreamOnImmutableArray() {
    (table<map<string>>)[] & readonly arr = [tbl];
    stream<table<map<string>> & readonly> strm = arr.toStream();

    stream<table<map<string>> & readonly> castedstrm = <stream<table<map<string>> & readonly>> strm;
}
```
I removed the table case, and ran all tests by building, and it was successful. https://github.com/Nadeeshan96/ballerina-lang/pull/6/files

Reply:
We have restricted usage of toStream() to arrays. Hence, +1 to remove the table case.

Reply:
+1